File: packages/keep-awake/android/src/main/java/io/capawesome/capacitorjs/plugins/keepawake/KeepAwake.java
package io.capawesome.capacitorjs.plugins.keepawake;

import android.view.WindowManager;
import androidx.annotation.NonNull;
import io.capawesome.capacitorjs.plugins.keepawake.classes.results.IsAvailableResult;
import io.capawesome.capacitorjs.plugins.keepawake.classes.results.IsKeptAwakeResult;

public class KeepAwake {

    @NonNull
    private final KeepAwakePlugin plugin;

    private boolean keptAwake = false;

    public KeepAwake(@NonNull KeepAwakePlugin plugin) {
        this.plugin = plugin;
    }

    public void allowSleep() {
        keptAwake = false;
        plugin
            .getActivity()
            .runOnUiThread(() -> plugin.getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON));
    }

    @NonNull
    public IsAvailableResult isAvailable() {
        return new IsAvailableResult(true);
    }

    @NonNull
    public IsKeptAwakeResult isKeptAwake() {
        return new IsKeptAwakeResult(keptAwake);
    }

    public void keepAwake() {
        keptAwake = true;
        plugin.getActivity().runOnUiThread(() -> plugin.getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON));
    }
}

File: packages/keep-awake/ios/Plugin/KeepAwake.swift
import Foundation
import UIKit

@objc public class KeepAwake: NSObject {
    private let plugin: KeepAwakePlugin

    init(plugin: KeepAwakePlugin) {
        self.plugin = plugin
    }

    @objc public func allowSleep(completion: @escaping () -> Void) {
        DispatchQueue.main.async {
            UIApplication.shared.isIdleTimerDisabled = false
            completion()
        }
    }

    @objc public func isAvailable() -> IsAvailableResult {
        return IsAvailableResult(available: true)
    }

    @objc public func isKeptAwake(completion: @escaping (IsKeptAwakeResult) -> Void) {
        DispatchQueue.main.async {
            let keptAwake = UIApplication.shared.isIdleTimerDisabled
            completion(IsKeptAwakeResult(keptAwake: keptAwake))
        }
    }

    @objc public func keepAwake(completion: @escaping () -> Void) {
        DispatchQueue.main.async {
            UIApplication.shared.isIdleTimerDisabled = true
            completion()
        }
    }
}
